From a296f52d216383cad8a2326daaa0541f2f0ede66 Mon Sep 17 00:00:00 2001
From: Alexandre Frade <kernel@xanmod.org>
Date: Sun, 8 Feb 2026 00:53:35 -0500
Subject: [PATCH 48/190] file: export file_close_fd() for GPL modules

Export file_close_fd() to GPL modules, prerequisite for
binder-as-module support.
---
 fs/file.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/fs/file.c b/fs/file.c
index 0a4f3bdb2dec..0046d0034bf1 100644
--- a/fs/file.c
+++ b/fs/file.c
@@ -879,10 +879,11 @@ struct file *file_close_fd(unsigned int fd)
 	file = file_close_fd_locked(files, fd);
 	spin_unlock(&files->file_lock);
 
 	return file;
 }
+EXPORT_SYMBOL(file_close_fd);
 
 void do_close_on_exec(struct files_struct *files)
 {
 	unsigned i;
 	struct fdtable *fdt;
-- 
2.52.0

